### PR TITLE
Support the fixable warnings and edit actions as Code Actions

### PR DESCRIPTION
**Pull request by rictic**
- [x] CHANGELOG.md has been updated

code actions

---
This change is [Reviewable](https://reviewable.io/reviews/polymer/polymer-editor-service/71)

### CODE REVIEW

File: src/language-server/language-server.ts [hunk 1]
```diff
@@ -174,6 +184,62 @@ export default class LanguageServer extends AutoDisposable {
...
+        return this.handleErrors(
+            this.executeApplyEditCommand(req.arguments as [WorkspaceEdit]),
+            undefined);
+      }
+    });
+  }
+
+  private async getCodeActions(req: CodeActionParams) {
+    const commands: Command[] = [];
+    if (req.context.diagnostics === []) {
```
Comment (valdrinkoshi) [inline src/language-server/language-server.ts]: This strict equality is always false.  Should it check for `.length` instead?

File: src/language-server/language-server.ts [hunk 2]
```diff
@@ -174,6 +184,62 @@ export default class LanguageServer extends AutoDisposable {
...
+    for (const warning of warnings) {
+      if ((!warning.fix && !warning.actions) ||
+          !isRangeInside(warning.sourceRange, range)) {
+        continue;
+      }
+      if (warning.fix) {
+        commands.push(this.createApplyEditCommand(
+            `Quick fix the '${warning.code}' warning`, warning.fix));
+      }
+      for (const action of warning.actions || []) {
```
Comment (valdrinkoshi) [inline src/language-server/language-server.ts]: nitty nit: avoid creating a new array here

File: src/language-server/language-server.ts [hunk 3]
```diff
@@ -174,6 +184,62 @@ export default class LanguageServer extends AutoDisposable {
...
+      // Currently we only support code actions on Warnings,
+      // so we can early-exit in the case where there aren't any.
+      return commands;
+    }
+    const warnings = await this._editorService.getWarningsForFile(
+        this.converter.getWorkspacePathToFile(req.textDocument));
+    const range = this.converter.convertLRangeToP(req.range, req.textDocument);
+    for (const warning of warnings) {
+      if ((!warning.fix && !warning.actions) ||
+          !isRangeInside(warning.sourceRange, range)) {
```
Comment (valdrinkoshi) [inline src/language-server/language-server.ts]: curiosity: how could this happen?

File: src/language-server/language-server.ts (hunk 2 above)
Comment (rictic) [reply src/language-server/language-server.ts]: Oh, and, done!

File: src/language-server/language-server.ts (hunk 3 above)
Comment (valdrinkoshi) [reply src/language-server/language-server.ts]: ah! Yeah the question was regarding the `isRangeInside` check, and wasn't clear that it's done to filter out warnings from other parts of the file. Would be great to have a comment explaining this 🙏

**Pull request merged**